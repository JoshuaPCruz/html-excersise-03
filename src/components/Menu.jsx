import React from 'react';
import '../css/components/ButtonMenu.css'
import '../css/components/Menu.css'

const Menu = () => {
  return (
    <ul className="Menu">
      <li className="Button-Menu-Active">Inicio</li>
      <li className="Button-Menu">Acerca</li>
      <li className="Button-Menu">Servicios</li>
      <li className="Button-Menu">Productos</li>
      <li className="Button-Menu">Contacto</li>
    </ul>
  );
};

export default Menu;