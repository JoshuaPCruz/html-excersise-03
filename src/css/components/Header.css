.Header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 13px 32px 32px;
}