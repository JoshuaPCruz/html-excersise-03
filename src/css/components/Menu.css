.Menu {
  display: flex;
}

.Menu li {
  display: inline-block;
  margin-left: 35px;
}

.Menu li:first-child {
  margin-left: 0;
}