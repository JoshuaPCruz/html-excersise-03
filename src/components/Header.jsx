import React from 'react';
import Menu from './Menu';
import logo from '../images/Logo.png';
import '../css/components/Header.css';

const Header = () => {
  return (
    <header className="Header">
      <a href="">
        <img src={logo} />
      </a>
      <Menu />
    </header>
  );
};

export default Header;