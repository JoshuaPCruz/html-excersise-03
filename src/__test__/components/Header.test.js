import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';
import Menu from '../../components/Menu';

describe('<Header />', () => {
  const header = shallow(<Header />);
  test('Render del componente Header', () => {
    expect(header.length).toEqual(1);
  });

  test('Render del <Menu /> en el <Header />', () => {
    expect(header.contains(<Menu />)).toBeTruthy();
  });
});

describe('Header Snapshot', () => {
  test('Comprobar la UI del componente Header', () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});