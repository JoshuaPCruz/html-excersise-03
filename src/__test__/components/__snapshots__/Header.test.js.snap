// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Comprobar la UI del componente Header 1`] = `
<header
  className="Header"
>
  <a
    href=""
  >
    <img
      src={Object {}}
    />
  </a>
  <ul
    className="Menu"
  >
    <li
      className="Button-Menu-Active"
    >
      Inicio
    </li>
    <li
      className="Button-Menu"
    >
      Acerca
    </li>
    <li
      className="Button-Menu"
    >
      Servicios
    </li>
    <li
      className="Button-Menu"
    >
      Productos
    </li>
    <li
      className="Button-Menu"
    >
      Contacto
    </li>
  </ul>
</header>
`;
